src/TicketMap/index.tsx

```diff
@@ -368,13 +370,38 @@ export default class TicketMap extends Component<Props & DefaultProps, State> {
 
   onClick = () => this.doSelect()
 
-  onTouchMove = () => this.setState({ isDragging: true })
+  onTouchStart = (event: React.TouchEvent<HTMLElement>) => {
+    if (event.changedTouches.length !== 1) {
+      return
+    }
+
+    const [ touchID ] = Object.getOwnPropertyNames(event.changedTouches)
+    const touch = event.changedTouches[touchID as any]
+
+    this.setState({
+      touchStarts: {
+        ...this.state.touchStarts,
+        [touchID]: {
+          x: touch.pageX,
+          y: touch.pageY
+        }
+      }
+    })
+  }
+
+  onTouchEnd = (event: React.TouchEvent<HTMLElement>) => {
+    if (event.changedTouches.length !== 1) {
+      return
+    }
+
+    const [ touchID ] = Object.getOwnPropertyNames(event.changedTouches)
+    const touch = event.changedTouches[touchID as any]
+    const touchStart = this.state.touchStarts[touchID]
 
-  onTouchEnd = ({ target }: React.TouchEvent<HTMLElement>) => {
-    if (!this.state.isDragging) {
-      this.doSelect(this.getSectionFromTarget(target as HTMLElement))
+    const translationDistance = Math.sqrt(Math.pow(touchStart.x - touch.pageX, 2) + Math.pow(touchStart.y - touch.pageY, 2))
+    if (translationDistance <= MAX_SELECT_TRANSLATION_DISTANCE) {
```

Comment:
you did manage to get a phone and test that this distance didn't cause any unexpected behavior?

Reply:
yep